packages/back-end/src/services/features.ts

```diff
@@ -393,3 +397,40 @@ export function getApiFeatureObj(
 
   return featureRecord;
 }
+
+/**
+ * Given the common {@link FeatureInterface} for both previous and next states, and the organization,
+ * will log an update event in the events collection
+ * @param organization
+ * @param previousState
+ * @param currentState
+ */
+export async function logFeatureUpdatedEvent(
+  organization: OrganizationInterface,
+  previousState: FeatureInterface,
+  currentState: FeatureInterface
+): Promise<string> {
+  // TODO: Add caching here since this rarely changes
+  const groupMap = await getSavedGroupMap(organization);
+
+  const previous = getApiFeatureObj(previousState, organization, groupMap);
+  const current = getApiFeatureObj(currentState, organization, groupMap);
+
+  const eventId = `event-${randomUUID()}`;
+  const payload: FeatureUpdatedNotificationEvent = {
+    event_id: eventId,
+    object: "feature",
+    event: "feature.updated",
+    data: {
+      current,
+      previous,
+    },
+  };
+
+  const emittedEvent = await createEvent(organization.id, payload);
+
+  const eventEmitter = getEventEmitterInstance();
+  eventEmitter.emit(EmittedEvents.EVENT_CREATED, emittedEvent.id);
```

Comment:
I would change this to one of the following.

Either
1. Emit the full `emittedEvent` object instead of the id (to avoid unnecessary mongo queries in the handlers)
2. OR use Agenda instead of an `EventEmitter` to process the event asynchronously

I would probably vote for option 2.  Handling the event will require several Mongo queries (e.g. get all webhooks for an org, get slack connection info) and this will grow over time as we add more types of handlers.  All of these queries introduce potential performance issues and failure points.  Agenda is similar to EventEmitter, but it adds fault tolerance and scalability by persisting to Mongo and having separate worker threads.

Reply:
I've removed the event emitter stuff and have created a class `EventNotifier` (which may need better naming). It creates a unique job for the event and enqueues it. The data stored is the `NotificationEventPayload<T,U,V>` which is handled by the `webHooksEventHandler` and, in the future, the commented out `slackEventHandler`. I've also added a `NotificationEventHandler` interface so we can implement that when we add new handlers.